src/connectors/connectStateResults.js

```diff
@@ -0,0 +1,41 @@
+import { _objectSpread } from '../util/polyfills';
+
+const connectStateResults = (renderFn, unmountFn = () => {}) => (
+  widgetParams = {}
+) => ({
+  $$type: 'vue.stateResults',
```

Comment:
Why is this the only VIS widget having the `$$type` key? (we're still using IS.js v3 here)

Reply:
I added the type in preparation for v4, so we don't need to add it later.

Reply:
It means that it's going to be inconsistent in this version.

(We should probably go for `avis` since it's `ais` in IS.js.)

Reply:
Avis is a regular word, so would confuse me. 

You're right that it's inconsistent, but I don't see the issue, since it isn't used anywhere. WDYT?

Reply:
Inconsistency brings confusion, there's no reasons to have this key in this VIS version :/

Besides, we rushed the decision to prefix it with `vue.`.

If we're at VIS 5 and look back at VIS 2 and see this key, we'll be confused: "What is this for?" "Did people rely on it already?" "We've never used this prefix elsewhere."

Reply:
That makes sense, do you suggest no $$type at all, also in v3?

Reply:
Yeah, let's not introduce it until we need it. We'll add `connectStateResults` in IS.js at some point anyway (perhaps with a better name?).

Reply:
https://github.com/algolia/vue-instantsearch/commit/fcf6469fff974bb4f218bbc53db6c426e32c2f5d